kernel/src/platform/mpu.rs

```diff
@@ -20,14 +18,41 @@ pub enum ExecutePermission {
     ExecutionNotPermitted = 0b1,
 }
 
+pub struct Region {
```

Comment:
The `empty` constructor is important because even when a region is marked "invalid" we still need to know the region number. This is not _merely_ a choice of who's responsible for the keeping track of the region number because regions can (and do) overlap, and their ordering makes a difference to who "wins".